djangosaml2idp/models.py

```diff
@@ -196,3 +196,14 @@ def resulting_config(self) -> str:
             config_as_str = f'Could not render config: {e}'
         # Some ugly replacements to have the json decently printed in the admin
         return mark_safe(config_as_str.replace("\n", "<br>").replace("    ", "&nbsp;&nbsp;&nbsp;&nbsp;"))
+
+
+class PersistentId(models.Model):
+    user = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
+    sp = models.ForeignKey(ServiceProvider, on_delete=models.CASCADE)
+    persistent_id = models.CharField("User Persistent Id for this SP", max_length=256)
+    created = models.DateTimeField(default=now)
+
+    class Meta:
+        unique_together = [["sp", "persistent_id"], ["sp", "user"]]
```

Comment:
Yes, since it is their recommendation at this point to do so, I agree it's a good idea to follow. This does mean we can no longer use Django 2.0 and 2.2 though. Since they are out of support by now, this does seem like a problem to me.

Reply:
Ah yeah I don't think we want to do that then.

Reply:
Why don't we leave it for now, and revisit in a few release versions where dependent systems are less likely to be using Django 2/2.2?

Reply:
My previous comment should have been Django 2.0 and *2.1*. Those are no longer supported. The UniqueConstraint has been added in 2.2 and this one is still supported as an LTS, next to the 3.x releases. So I'd prefer to start using that already.

Reply:
Done